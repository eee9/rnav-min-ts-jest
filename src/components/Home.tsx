/* eslint-disable @typescript-eslint/no-unused-vars */
import * as React from 'react';
import {Button, Text, View, StyleSheet, BackHandler} from 'react-native';

import {IDrawer} from '../types';

import {cc, J} from '../libs/mxlib';
export interface IHome {
  runVer: string;
  message: string;
}

export const Home = (props: IDrawer & IHome) => {
  //cc('In Home(). props =>');
  //cc(J(props));
  return (
    <View style={styles.HomeStyle}>
      <Text>{props.runVer}, Drawer. Home screen</Text>
      <Text>message: [{props.message}]</Text>
      <Text> </Text>
      <Button title="Go to Profile" onPress={() => props.navigation.navigate('Profile')} />
      <Text> </Text>
      <Button title="Go to Settings" onPress={() => props.navigation.navigate('Settings')} />
      <Text> </Text>
      <Button title="Exit" onPress={BackHandler.exitApp} />
    </View>
  );
};

const styles = StyleSheet.create({
  HomeStyle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'cyan',
  },
});
