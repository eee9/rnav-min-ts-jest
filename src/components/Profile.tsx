import * as React from 'react';
import {Button, Text, View, StyleSheet} from 'react-native';

import {IDrawer} from '../types';

export const Profile = ({navigation}: IDrawer) => {
  return (
    <View style={styles.ProfileStyle}>
      <Text>Profile screen</Text>
      <Text> </Text>
      <Button title="Go to Settings" onPress={() => navigation.navigate('Settings')} />
      <Text> </Text>
      <Button title="Go back" onPress={() => navigation.goBack()} />
    </View>
  );
};

const styles = StyleSheet.create({
  ProfileStyle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'lightyellow',
  },
});
