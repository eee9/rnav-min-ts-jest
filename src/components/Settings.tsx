import * as React from 'react';
import {Button, Text, View, StyleSheet} from 'react-native';

import {IDrawer} from '../types';

export const Settings = ({navigation}: IDrawer) => {
  return (
    <View style={styles.SettingsStyle}>
      <Text>Settings screen</Text>
      <Text> </Text>
      <Button title="Go back" onPress={() => navigation.goBack()} />
      <Text> </Text>
      <Button title="Go Home" onPress={() => navigation.navigate('Home')} />
    </View>
  );
};

const styles = StyleSheet.create({
  SettingsStyle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'lightgreen',
  },
});
