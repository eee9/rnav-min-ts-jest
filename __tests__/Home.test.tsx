import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';

import {Home, IHome} from '../src/components/Home';

const navigation = null;
const props: IHome = {
  runVer: 'r00, L00',
  message: 'Message for Home.test!',
};

describe('Home.tsx render', () => {
  it('@testing-library/react-native renders correctly', () => {
    const {toJSON} = render(<Home {...props} {...navigation} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
