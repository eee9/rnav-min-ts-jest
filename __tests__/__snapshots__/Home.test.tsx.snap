// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.tsx render @testing-library/react-native renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "cyan",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    r00, L00
    , Drawer. Home screen
  </Text>
  <Text>
    message: [
    Message for Home.test!
    ]
  </Text>
  <Text>
     
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go to Profile
      </Text>
    </View>
  </View>
  <Text>
     
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go to Settings
      </Text>
    </View>
  </View>
  <Text>
     
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Exit
      </Text>
    </View>
  </View>
</View>
`;
